Sources/Markdown/Parser/BlockDirectiveParser.swift

```diff
@@ -265,12 +265,12 @@ struct TrimmedLine {
     }
 
     /// - parameter untrimmedText: ``untrimmedText``
-    init(_ untrimmedText: Substring, source: URL?, lineNumber: Int?, parseIndex: Substring.Index? = nil) {
+    init(_ untrimmedText: Substring, source: URL?, lineNumber: Int?, parseIndex: Substring.Index? = nil, startParseIndex: Substring.Index? = nil) {
         self.untrimmedText = untrimmedText
         self.source = source
         self.parseIndex = parseIndex ?? untrimmedText.startIndex
         self.lineNumber = lineNumber
-        self.startParseIndex = self.parseIndex
+        self.startParseIndex = startParseIndex ?? self.parseIndex
```

Comment:
This is useful, but i wonder if it would be better if the `location` property were able to scan backwards to the previous newline character (or the beginning of the string)? That way these calculations would be much more reliable.

Reply:
Should we only lazily scan once for the first time location is accessed? Or should we compute it every time? 

Overall, I'm -1 on back scan and delete startParseIndex property.

Since Markdown is a low level library I think performance is important in some way. And that's why we prefer to use index almost everywhere when parsing.

Reply:
Part of me thinks that caching the `startParseIndex` in that situation wouldn't make much of a difference, since `TrimmedLine`s are generally created and dropped a lot in the parsing logic. And because of that, it's better for performance that we can use the index we already have on hand. I'll drop this comment.